presto-spi/src/main/java/com/facebook/presto/spi/ConnectorSplit.java

```diff
@@ -37,8 +39,18 @@
      */
     List<HostAddress> getPreferredNodes(NodeProvider nodeProvider);
 
+    /**
+     * This method returns a raw object.
+     * <p> Instead use {@link #getInfoMap()} method which returns a <pre>{@code Map<String, String>}</pre>
+     */
+    @Deprecated
     Object getInfo();
 
+    default Map<String, String> getInfoMap()
+    {
+        return new HashMap<>();
```

Comment:
Should we return null here to indicate the connector hasn't implemented this?  It may more cleanly indicate lack of support than empty map (which may be reasonable to return if there's no useful information being returned from the connector).

Reply:
I see your point.
Offering a counter argument.
If there is no useful information returned from the connector, they need not override this method and can rely on the default implementation. If they override this method it should be to add something in the map. The advantage of returning an empty map is that we can get rid of all the null checks.

Reply:
My worry is a situation where for some reason someone specifies an empty map for the map version, but a non-empty object for the object version.  It would be great to fail loudly if we're in an inconsistent state like that.  If we use an empty map as the default, it seems harder to know that.